### PR TITLE
OSRA427 - Integration tests for CRUD in Admin Users functionality

### PR DESCRIPTION
**Pull request by gmunumel**
The `aa` functionality have a `delete` link for admin users. In `hq` that link doesn't exist, thus I'm not testing that. 

This PR is mainly a refactor of three files: `admin_user_authentificatio_spec.rb`, `create_admin_user_spec.rb` and `edit_admin_user_spec.rb` into `crud_admin_user_spec.rb`. 

JIRA activity: https://osraav.atlassian.net/browse/OSRA-427

### CODE REVIEW

File: spec/features/hq/crud_admin_user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    i_sign_out
+    i_sign_in_with_credentials('newadmin.user@example.com', 'NewPassword')
+    and_i_should_see "Logged in successfully"
+  end
+
+  def and_i_should_not_have_access_to_restricted_application_pages
+    visit hq_root_path
+    expect(current_path).to eq new_hq_admin_user_session_path
+  end
+end
```
Comment (razorcd) [inline spec/features/hq/crud_admin_user_spec.rb]: lets rename this `admin_user_CRUD_and_authentication_spec.rb`
Also we should check that index has NO delete button for `current_user` and has delete buttons for other admin_users. And test the `delete` behavior too.

File: spec/features/hq/crud_admin_user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    click_button 'Create Admin User'
+    i_sign_out
+    i_sign_in_with_credentials('newadmin.user@example.com', 'Password')
+    and_i_should_see "Logged in successfully"
+  end
+
+  scenario 'After resetting their password Admin Users should be able to log in with the new credentials' do
+    i_sign_in_as_admin
+    admin_user_to_reset = FactoryGirl.create :admin_user
+    visit edit_hq_admin_user_path(admin_user_to_reset)
```
Comment (razorcd) [inline spec/features/hq/crud_admin_user_spec.rb]: better go to index and click "Edit"  for correct admin_user

File: spec/features/hq/crud_admin_user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  scenario 'Logging in only possible with correct credentials' do
+    i_sign_in_with_wrong_credentials
+    and_i_should_be_on :sign_in_page
+    and_i_should_see "Invalid email or password"
+  end
+
+  scenario 'After creation new Admin Users should be able to log in' do
+    i_sign_in_as_admin
+    visit new_hq_admin_user_path
```
Comment (razorcd) [inline spec/features/hq/crud_admin_user_spec.rb]: better go to index and click "New"

**Comment by razorcd** [comment]
please rebase and leave a comment here too

**Comment by gmunumel** [comment]
I have forgot to do this. I'll try to do it this weekend. Sorry.

**Comment by gmunumel** [comment]
Running my tests locally without problem. Semaphore doesn't like Consuelo lol (Consuelo is a Spanish name) :P

**Comment by razorcd** [comment]
haha, yes, I see. Forgot to add some escaping on`'` again :)
I'll fix that

**Comment by razorcd** [comment]
oh no, you need to rebase this again. It has conflicts after I merged the other PR

**Comment by gmunumel** [comment]
Cool. I'll do it this night. :+1:

**Pull request merged**